pkg/controllers/deprovisioning/helpers.go

```diff
@@ -164,8 +204,14 @@ func GetCandidates(ctx context.Context, cluster *state.Cluster, kubeClient clien
 		cn, e := NewCandidate(ctx, kubeClient, clk, n, provisionerMap, provisionerToInstanceTypes)
 		return cn, e == nil
 	})
+	for _, c := range candidates {
+		// If machine is prohibited by a Machine Disruption Gate, emit event and skip deprovisioning.
+		if prohibited.Has(c.Name()) {
```

Comment:
```suggestion
		if prohibited.Has(c.Machine.Name) {
```

This function `c.Name()` gets an "effective name" for the state node. This will be the node name if both the machine and the node exist in the cluster state. 

Based on this usage, I'm starting to question whether we should pull out this `Name()` function since it has the potential to be confusing